internal/sanitize/sts.go

```diff
@@ -120,6 +121,26 @@ func (s *StatefulSet) checkContainers(ctx context.Context, st *appsv1.StatefulSe
 	}
 }
 
+// checkSameNode verifies if all replicas of the StatefulSet are running on
+// the same node.
+func (s *StatefulSet) checkSameNode(ctx context.Context, st *appsv1.StatefulSet) {
+	if *st.Spec.Replicas <= 1 {
+		return
+	}
+
+	nodeMap := make(map[string]int)
```

Comment:
Don't think we need to count here ie if the node is already in the map the pods are collocated
```go
s.ListPodsBySelector(st.Namespace, st.Spec.Selector
make(map[string]struct{}, len(oo))
if _, ok := nodeMap[pod.Spec.NodeName]; ok {
   s.AddCode(...)
   return
```

That said I don't think these are valid general checks ie ES multiple shards can land on the same node? Thus that would raise a false positive??